cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/realtime/kafka/Kafka08SimpleApiConsumer.java

```diff
@@ -238,13 +238,13 @@ public void destroy() {
   }
 
   @Override
-  protected void processMessage(String key, ByteBuffer payload, Emitter<StructuredRecord> emitter) {
+  protected void processMessage(String key, byte[] payload, Emitter<StructuredRecord> emitter) {
```

Comment:
I was thinking to just pass the bytes array to make sure dowan stream code to avoid consider position and remaining of the ByteBuffer instead of calling ByteBuffer.array by mistake.
On the other hand, we could just assume dowan stream sinks knew how to deal with ByteBuffer properly.